engine/consensus/matching/engine.go

```diff
@@ -654,17 +640,52 @@ func (e *Engine) sealResult(result *flow.ExecutionResult, final *flow.Header) er
 	return nil
 }
 
-func (e *Engine) clearPools(sealedIDs []flow.Identifier) {
+// collectAggregateSignatures collects approver signatures and identities per
+// chunk
+// TODO: needs testing
+func (e *Engine) collectAggregateSignatures(result *flow.ExecutionResult) []flow.AggregatedSignature {
+	resultID := result.ID()
+	signatures := make([]flow.AggregatedSignature, 0, len(result.Chunks))
+
+	for _, chunk := range result.Chunks {
+		// get approvals for result with chunk index
+		approvals := e.approvals.ByChunk(resultID, chunk.Index)
+		if approvals == nil {
+			continue
+		}
```

Comment:
⚠️ potentially collecting approvals from non-assigned verifiers:

* In https://github.com/onflow/flow-go/blob/4b7c696b01e7881db12631f7a49d6ed4123c1e54/engine/consensus/matching/engine.go#L586-L595 you match only the approvals from assigned verifiers
* here, you collect all approvals, even if the verifier is _not_ assigned to the respective chunk. This is not compliant with the protocol
* Furthermore, note that there is no check for cryptographic correctness of the approval. For now, we might put invalid approvals in the mempool. We need to ensure that invalid approvals are not included here.